File: financeiro-api/src/main/java/br/com/iftm/financeiroapi/model/repository/EntryRepository.java
package br.com.iftm.financeiroapi.model.repository;

import br.com.iftm.financeiroapi.model.domain.Entry;
import org.springframework.stereotype.Component;

import java.io.IOException;
import java.util.List;

@Component
public interface EntryRepository {

    void save(Entry entry) throws IOException;

    void update(Entry entry) throws IOException;

    void delete(String id) throws IOException;

    Entry findById(String id) throws IOException;

    List<Entry> findAll() throws IOException;

}

File: financeiro-api-kotlin/src/main/kotlin/br/com/iftm/financeiroapi/model/repository/EntryRepository.kt
package br.com.iftm.financeiroapi.model.repository

import br.com.iftm.financeiroapi.model.domain.Entry
import org.springframework.stereotype.Component

@Component
interface EntryRepository {

    fun save(entry: Entry)

    fun update(entry: Entry)

    fun delete(id: String)

    fun findById(id: String): Entry?

    fun findAll(): List<Entry>

}





